pkg/client/client.go

```diff
@@ -864,6 +889,59 @@ func (c *Client) WaitForPrometheus(ctx context.Context, p *monv1.Prometheus) err
 	return nil
 }
 
+func (c *Client) ValidatePrometheus(ctx context.Context, promNsName types.NamespacedName) error {
+	var validationError error
+
+	pollErr := wait.Poll(10*time.Second, 5*time.Minute, func() (bool, error) {
+
+		p, err := c.GetPrometheusByNsName(ctx, promNsName)
+		if err != nil {
+			klog.V(4).Info("validate prometheus failed to get prometheus: ", err)
+			return false, err
+		}
+
+		avail, err := prometheusConditonForType(p.Status.Conditions, monv1.PrometheusAvailable)
+		if err != nil {
+			validationError = NewUnavailableError(err.Error())
+			return false, nil
+		}
+
+		if avail.Status == monv1.PrometheusConditionTrue {
+			return true, nil // terminate if available
+		}
+
+		// return reason for failure as state-errors - Degraded/Unavailable
+		errs := []error{}
+		msg := fmt.Sprintf("%s: %s", avail.Reason, avail.Message)
+
+		newStateError := NewDegradedError
+		if avail.Status == monv1.PrometheusConditionFalse {
+			// prometheus not available is a Degraded and Unavailable error
+			errs = append(errs, NewDegradedError(msg))
+			newStateError = NewUnavailableError
+		}
+
+		errs = append(errs, newStateError(msg))
+		validationError = apiutilerrors.NewAggregate(errs)
+		return false, nil
+	})
+
+	if pollErr == wait.ErrWaitTimeout {
```

Comment:
When we timeout retrieving the prometheus object, shouldn't this create an UnknownStateError?

Reply:
Good catch!! 
I think it makes sense to mark both Degraded and Available as `Unknown` WDYT? 
cc: @simonpasquier

Reply:
I think that makes sense yes. Though based on my comment further up, maybe it makes sense to change this function to check for the `Available` and `Reconciled` conditions of Prometheus separately and deduce from that which errors we want to pass?

Reply:
I feel that  from the point of view of CMO, failure to get the state information of `prometheus` should be communicated to back to the user as  `Degraded` and `Available` (of CMO) as `unknown`
WDYT?

Reply:
I have updated the `validationError` to return appropriate state errors, so returning `validationError` should suffice.

Reply:
Will add degraded as well to the list of errors if we timeout and change the decision based on feedback later.